File: apps/api/src/lib/responses.ts
import type { Context } from 'hono'

export function ok<T>(c: Context, data: T, status: 200 | 201 = 200) {
  return c.json({ success: true as const, data }, status)
}

export function fail(c: Context, error: string, status: 400 | 401 | 403 | 404 | 409 | 500 = 400) {
  return c.json({ success: false as const, error }, status)
}

File: backend/src/clawbuddy/lib/responses.py
"""Standardized JSON response helpers.

Replaces: apps/api/src/lib/responses.ts
"""

from __future__ import annotations

from typing import Any, Literal, TypeVar

from fastapi.responses import JSONResponse

T = TypeVar("T")


def ok(data: Any, status_code: Literal[200, 201] = 200) -> JSONResponse:
    """Return a success JSON response."""
    return JSONResponse(
        content={"success": True, "data": data},
        status_code=status_code,
    )


def fail(
    error: str,
    status_code: Literal[400, 401, 403, 404, 409, 500] = 400,
) -> JSONResponse:
    """Return an error JSON response."""
    return JSONResponse(
        content={"success": False, "error": error},
        status_code=status_code,
    )
